Mathlib/Analysis/Convex/Cone/Proper.lean

```diff
@@ -30,77 +32,107 @@ The next steps are:
+protected nonrec lemma smul_mem (C : ProperCone R E) (hx : x ∈ C) (hr : 0 ≤ r) : r • x ∈ C :=
+  C.smul_mem ⟨r, hr⟩ hx
+
+section T1Space
+variable [T1Space E]
+
+lemma mem_bot : x ∈ (⊥ : ProperCone R E) ↔ x = 0 := .rfl
+
+@[simp, norm_cast] lemma toPointedCone_bot : (⊥ : ProperCone R E).toPointedCone = ⊥ := rfl
+
+end T1Space
+
+/-- The closure of image of a proper cone under a `ℝ`-linear map is a proper cone. We
+use continuous maps here so that the comap of f is also a map between proper cones. -/
+abbrev comap (f : E →L[R] F) (C : ProperCone R F) : ProperCone R E :=
+  ClosedSubmodule.comap (f.restrictScalars R≥0) C
 
-protected theorem nonempty (K : ProperCone 𝕜 E) : (K : Set E).Nonempty :=
-  ⟨0, by { simp_rw [SetLike.mem_coe, ← ProperCone.mem_coe, Submodule.zero_mem] }⟩
+@[simp] lemma comap_id (C : ProperCone R F) : C.comap (.id _ _) = C := rfl
 
-protected theorem isClosed (K : ProperCone 𝕜 E) : IsClosed (K : Set E) :=
-  K.isClosed'
+@[simp] lemma coe_comap (f : E →L[R] F) (C : ProperCone R F) : (C.comap f : Set E) = f ⁻¹' C := rfl
+
+lemma comap_comap (g : F →L[R] G) (f : E →L[R] F) (C : ProperCone R G) :
+    (C.comap g).comap f = C.comap (g.comp f) := rfl
+
+lemma mem_comap {C : ProperCone R F} {f : E →L[R] F} : x ∈ C.comap f ↔ f x ∈ C := .rfl
+
+variable [ContinuousAdd F] [ContinuousConstSMul R F]
+
+/-- The closure of image of a proper cone under a linear map is a proper cone.
+
+We use continuous maps here to match `ProperCone.comap`. -/
+abbrev map (f : E →L[R] F) (C : ProperCone R E) : ProperCone R F :=
+  ClosedSubmodule.map (f.restrictScalars R≥0) C
+
+@[simp] lemma map_id (C : ProperCone R F) : C.map (.id _ _) = C := ClosedSubmodule.map_id _
+
+@[simp, norm_cast]
+lemma coe_map (f : E →L[R] F) (C : ProperCone R E) :
+    C.map f = (C.toPointedCone.map (f : E →ₗ[R] F)).closure := rfl
+
+@[simp]
+lemma mem_map {f : E →L[R] F} {K : ProperCone R E} {y : F} :
+    y ∈ K.map f ↔ y ∈ (PointedCone.map (f : E →ₗ[R] F) ↑K).closure := .rfl
 
 end Module
 
 section PositiveCone
+variable {E : Type*} [AddCommGroup E] [TopologicalSpace E] [Module R E] [PartialOrder E]
+  [IsOrderedAddMonoid E] [OrderedSMul R E] [OrderClosedTopology E] {x : E}
 
-variable (𝕜 E)
-variable [Semiring 𝕜] [PartialOrder 𝕜] [IsOrderedRing 𝕜]
-  [AddCommGroup E] [PartialOrder E] [IsOrderedAddMonoid E] [Module 𝕜 E] [OrderedSMul 𝕜 E]
-  [TopologicalSpace E] [OrderClosedTopology E]
-
+variable (R E) in
 /-- The positive cone is the proper cone formed by the set of nonnegative elements in an ordered
 module. -/
-def positive : ProperCone 𝕜 E where
-  toSubmodule := PointedCone.positive 𝕜 E
+@[simps!]
+def positive : ProperCone R E where
+  toSubmodule := PointedCone.positive R E
   isClosed' := isClosed_Ici
 
-@[simp]
-theorem mem_positive {x : E} : x ∈ positive 𝕜 E ↔ 0 ≤ x :=
-  Iff.rfl
-
-@[simp]
-theorem coe_positive : ↑(positive 𝕜 E) = ConvexCone.positive 𝕜 E :=
-  rfl
+@[simp] lemma mem_positive : x ∈ positive R E ↔ 0 ≤ x := .rfl
```

Comment:
You might also want the lemma about the coercion to PointedCone or to ConvexCone (as was there before) `(positive R E : PointedCone R E) = .positive R E` (currently you're only generating the lemma about coercing to Set E).